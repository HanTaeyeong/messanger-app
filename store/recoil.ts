import { atom, selector, useRecoilState } from 'recoil';

export const weekList: String[] = ['Sunday', 'Monday', 'Tuesday', 'Wendsday', 'Thursday', 'Fryday', 'Saturday']

export const testDate = atom<Number>({
    key: 'testDate',
    default: 0,
});

export const testWeekName = selector<String>({
    key: 'testWeekName',
    get: ({ get }) => weekList[(+get(testDate) % 7) | 0] || 'Sunday',
    set: ({ set }) => { }
});
