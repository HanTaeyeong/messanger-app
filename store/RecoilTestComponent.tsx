import React from 'react'
import { useRecoilState } from 'recoil'
import { testDate, testWeekName } from './recoil'


export default function RecoilTestComponent() {
  const [date, setDate] = useRecoilState(testDate)
  const [weekName, setWeekName] = useRecoilState(testWeekName)
  
  const onClick = () => {
    setDate(1)
  }
  
  return (
    <div>
      <head className='date'>{date}</head>
      <div className='week-name'>{weekName}</div>
      <button className="test-button" onClick={onClick}>increase</button>
    </div>
  )
}
