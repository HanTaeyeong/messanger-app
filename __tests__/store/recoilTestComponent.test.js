import React from 'react'
import RecoilTestComponent from '../../store/RecoilTestComponent'
import { RecoilRoot } from 'recoil'


import { create, act } from 'react-test-renderer'

describe('recoil test', () => {
    let recoilComponent;

    beforeEach(async () => {
        recoilComponent = await create(
            <RecoilRoot>
                <RecoilTestComponent />
            </RecoilRoot>)
    })

    test('render test', () => {
        expect(recoilComponent).toBeTruthy();
        expect(recoilComponent.toJSON()).toMatchSnapshot();
    })

    test('date value changes well after click', async () => {
        const root = recoilComponent.root;
        let date = await root.findByProps({ className: "date" })
        const button = await root.findByProps({ className: "test-button" })

        expect(date.props.children).toBe(0)
        act(button.props.onClick);

        expect(date.props.children).toBe(1)
    })

    test('weekName value changes well after click', async () => {
        const root = recoilComponent.root;

        let weekName = await root.findByProps({ className: 'week-name' })
        const button = await root.findByProps({ className: "test-button" })

        expect(weekName.props.children).toBe('Sunday')

        act(button.props.onClick);

        expect(weekName.props.children).toBe('Monday')
    })





})