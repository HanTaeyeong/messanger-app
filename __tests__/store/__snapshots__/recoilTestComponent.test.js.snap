// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recoil test render test 1`] = `
<div>
  <head
    className="date"
  >
    0
  </head>
  <div
    className="week-name"
  >
    Sunday
  </div>
  <button
    className="test-button"
    onClick={[Function]}
  >
    increase
  </button>
</div>
`;
